projects/ngx-konva/src/lib/directives/ko-transform.directive.ts

```diff
@@ -45,8 +58,19 @@ export class KoTransformDirective implements OnInit, OnDestroy {
     this.node.off('transformstart', this.onTransformStarterListener);
     this.node.off('transformend', this.onTransformEndListener);
     this.node.off('transform', this.onTranformListener);
+    this.removeTransformer();
+  }
+
+  addTransformer() {
+    this.transformer = new Transformer({...this.koTransformOptions, nodes: [this.node]});
+    this.node.getLayer()?.add(this.transformer);
```

Comment:
Here you can retrieve the parent layer by adding KoLayerComponent in the DI constructor

Reply:
Not sure if there is really any real world difference but I've switched to DI KoLayerComponent.layer as a matter of style.